### ISSUE TITLE
Would it also work with an T230?

### ISSUE THREAD

**Issue by PanicMan** [psvanstrom/esphome-uh50reader#2]
Hello,
thank a lot for the Project!
I'm trying already since a long time to read out my L&G UH T230 Meter:
![grafik](https://user-images.githubusercontent.com/4090506/213791848-a5a7ee22-24c4-4de4-b10c-0409c5bbf38e.png)
https://www.manualslib.com/manual/1590596/LandisPlusgyr-Ultraheat-T230.html
I'm using that head: https://www.ebay.de/itm/353940190756
I tried also your code, first no change, then I add additional line to the wakeup sequence, no difference, TX to D4, RX to RX, the sequence is send, I can see it with my handy-cam.
What do you think, should it work with my Meter too? And with my Head? 
I also tried to wake up the Meter with the Button, also no response... Anything else what I can try? 
Thanks, Jack

**Comment by psvanstrom** [comment]
Hi!

I'm not sure to be honest, but since it has an optical interface and supports IEC 62056-21 I'm sure it should be able to get it to work.

Perhaps the starting wake-up sequence is different? In [this](https://github.com/lvzon/dsmr-p1-parser/blob/master/doc/IEC-62056-21-notes.md#wake-up) project the mention a start sequence of 65 zero-characters, perhaps try to modify the array at https://github.com/psvanstrom/esphome-uh50reader/blob/main/uh50reader.h#L27 and see if that has any effect?

**Comment by PanicMan** [comment]
Yeah, thanks, I can try that... Also that part that I have to wait some time is strange:
> Especially battery powered meters require a wake-up sequence to be sent before further communication on the optical IEC 62056-21 interface. In most cases, sending a string of 65 zero-characters ('\0') should do the trick, followed by a **pause of 1.5 seconds**. If serial data is sent asynchronously, make sure you also wait for the data to be sent, which probably requires a total **waiting time of 2.7 seconds** after sending the wake-up string.

I'm not sure, but it seems your reader is listening right after sending the wakeup. Should I wait here for an specific time or just wait for that "available()" flag? And if wait, how, is there any "sleep(ms);" command?

**Comment by falkio8** [comment]
Someone has working scripts for the T330 at https://www.photovoltaikforum.com/thread/188234-landis-gyr-ultraheat-t230-w%C3%A4rmez%C3%A4hler-mit-trct5000-und-esphome-wemos-d1-mini-aus/?postID=4221968#post4221968

Is there a way to implement those to make them work somehow with an integration to use it in HA?

**Comment by psvanstrom** [comment]
Hopefully someone with a T330 can test/implement support but since I don't have one I can't do that.
